internal/registry/git.go

```diff
@@ -0,0 +1,37 @@
+package registry
+
+import (
+	"bytes"
+	"os"
+	"text/template"
+
+	"github.com/pkg/errors"
+)
+
+func CreateGitCommit(b Buildpack, username string, registryCache Cache) error {
+	_, err := os.Stat(registryCache.Root)
+	if err != nil {
+		if os.IsNotExist(err) {
+			err = registryCache.CreateCache()
+			if err != nil {
+				return errors.Wrap(err, "creating registry cache")
+			}
+		}
+	}
+
+	commitTemplate, err := template.New("buildpack").Parse(GitCommitTemplate)
+	if err != nil {
+		return err
```

Comment:
nit: I prefer to wrap my errors whenever possible, to provide more helpful context on the error. Something like
```suggestion
		return errors.Wrap(err, "creating template")
```
(here, as well as in the next few errors) would be great